src/components/EditorWidgets/List/ListControl.js

```diff
@@ -173,6 +179,23 @@ export default class ListControl extends Component {
     });
   }
 
+  handleCollapseAllToggle = (e) => {
+    e.preventDefault();
+    const { value } = this.props;
+    const { allItemsCollapsed } = this.state;
+    const itemsCount = value ? value.size : 0;
+    let { itemsCollapsed } = this.state;
+
+    for (let i = 0; i < itemsCount; i++) {
+      itemsCollapsed = itemsCollapsed.set(i, !allItemsCollapsed);
+    }
```

Comment:
These last three lines can be replaced with something like 
```js
const newItemsCollapsed = Array(itemsCollapsed.length).fill(!allItemsCollapsed);
```

Reply:
Good point! The button updates it's state accordingly now.

But I sticked with the forloop, because I couldn't find a more slick solution. With the loop I can simultaneously create and set the list elements.

Reply:
This can remain as-is - @erquhart pointed out that this isn't as simple as I thought and the for-loop is a pretty simple solution here.